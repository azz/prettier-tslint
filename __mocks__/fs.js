import path from "path";

const actual = require.requireActual("fs");

let writes = {};

export const statSync = jest.fn(actual.statSync);
export const existsSync = jest.fn(actual.existsSync);

export const writeFileSync = jest.fn((filePath, content) => {
  const relative = path.relative(process.cwd(), filePath);
  writes[relative] = content;
});

export const readFileSync = jest.fn(filePath => {
  const relative = path.relative(process.cwd(), filePath);
  if (writes[relative]) {
    return writes[relative];
  }
  return actual.readFileSync(filePath, "utf8");
});

export const __clear = () => {
  writes = {};
  existsSync.mockClear();
  statSync.mockClear();
  writeFileSync.mockClear();
  readFileSync.mockClear();
};

export default {
  __clear,
  existsSync,
  statSync,
  readFileSync,
  writeFileSync,
};
