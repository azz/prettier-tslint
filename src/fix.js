import runTsLint from "./run-tslint";
import runPrettier from "./run-prettier";

const fix = filePath => {
  const prettierCheck = runPrettier(filePath, true);
  const tslintCheck = runTsLint(filePath, true);
  return prettierCheck && tslintCheck;
};

export default fix;
