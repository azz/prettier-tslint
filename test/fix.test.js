jest.mock("fs");

import fix from "../src/fix";
import fs from "fs";

const lastCall = fn => {
  return fn.mock.calls[fn.mock.calls.length - 1];
};

describe("fix()", () => {
  beforeEach(() => {
    fs.__clear();
  });

  test("good format, bad lint writes fix to disk", () => {
    fix("test/fixture/good-format-bad-lint.ts");
    expect(lastCall(fs.writeFileSync)).toMatchSnapshot();
  });

  test("bad format, good lint writes fix to disk", () => {
    fix("test/fixture/bad-format-good-lint.ts");
    expect(lastCall(fs.writeFileSync)).toMatchSnapshot();
  });

  test("bad format, bad lint writes fix to disk", () => {
    fix("test/fixture/bad-format-bad-lint.ts");
    expect(lastCall(fs.writeFileSync)).toMatchSnapshot();
  });

  test("good format, good lint writes doesn't write to disk", () => {
    fix("test/fixture/good-format-good-lint.ts");
    expect(fs.writeFileSync).toHaveBeenCalledTimes(0);
  });
});
