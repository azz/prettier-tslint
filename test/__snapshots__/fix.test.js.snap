// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fix() bad format, bad lint writes fix to disk 1`] = `
Array [
  "test/fixture/bad-format-bad-lint.ts",
  "throw new Error(\\"no-string-throw\\");
",
]
`;

exports[`fix() bad format, good lint writes fix to disk 1`] = `
Array [
  "test/fixture/bad-format-good-lint.ts",
  "throw new Error(\\"no-string-throw\\");
",
]
`;

exports[`fix() good format, bad lint writes fix to disk 1`] = `
Array [
  "test/fixture/good-format-bad-lint.ts",
  "throw new Error(\\"no-string-throw\\");
",
]
`;
